README.md

```diff
@@ -2,23 +2,95 @@
 
 ## What is it?
 
-Here's a very technical definition from [Wikipedia](https://en.wikipedia.org/wiki/Finite-state_machine)
+> There are many variants in definitions of finite state machines, depending on
+  whether one views them as language acceptors/generators (automata), or machines that react with their environment and produce output (eg. Mealy and
+  Moore machines, I/O automata), or logical models (Kripke structures) etc.
+> [...] In practice, several extensions are useful to enhance the expressive power and
+to describe more complex systems more efficiently
+
+*Source: [Hierarchical State Machines](https://link.springer.com/content/pdf/10.1007/3-540-44929-9_24.pdf), Mihalis Yannakakis, Bell Laboratories, 2000*
+  
+According to the extensions they incorporate, the following types of state machines are commonly distinguished:
+ 
+| Extension | Terminology | Characteristics |
+|:---|:---|:---|
+| - | basic finite state machine ([FSM]((https://en.wikipedia.org/wiki/Finite-state_machine)) | inputs, states, transitions |
```

Comment:
> ([FSM]((

Additional parenthesis after the `]` messes up the rendering for the closing parenthesis at the end of this line :)